external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/SerializableCredentialsProvider.scala

```diff
@@ -83,3 +84,146 @@ private[kinesis] final case class STSCredentialsProvider(
     }
   }
 }
+
+@InterfaceStability.Stable
+object SerializableCredentialsProvider {
```

Comment:
hmm. This isn't a great name for a user facing API, the user shouldn't have to care about if the provider is serializable or not, that's an implementation detail.

I understand your concerns with the `AWSCredentials` name collisions. However, I think it's the best name there is.